File: round1_full_pack_optionA_solutions/H3_Merge_K_Sorted_Arrays/Solution/Solution.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        if (!sc.hasNextInt()) return;
        int K = sc.nextInt();

        ArrayList<Integer> all = new ArrayList<>();

        // Read each array
        for (int i = 0; i < K; i++) {
            int n = sc.nextInt();
            for (int j = 0; j < n; j++) {
                all.add(sc.nextInt());
            }
        }

        // Sort
        Collections.sort(all);

        // Print
        for (int i = 0; i < all.size(); i++) {
            if (i > 0) System.out.print(" ");
            System.out.print(all.get(i));
        }
        System.out.println();

        sc.close();
    }
}

File: round1_full_pack_optionA_solutions/H3_Merge_K_Sorted_Arrays/Solution/Solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int K;
    if (!(cin >> K)) return 0;

    vector<int> all;

    // Read each array
    for (int i = 0; i < K; i++) {
        int n;
        cin >> n;
        for (int j = 0; j < n; j++) {
            int x;
            cin >> x;
            all.push_back(x);
        }
    }

    // Sort
    sort(all.begin(), all.end());

    // Print
    for (int i = 0; i < all.size(); i++) {
        if (i > 0) cout << " ";
        cout << all[i];
    }
    cout << endl;

    return 0;
}
